packages/scripts.nix

```diff
@@ -425,30 +425,32 @@
   # Usage: get-logs $namespaceFile
   get-logs = writeShellApplication {
     name = "get-logs";
-    runtimeInputs = with pkgs; [ kubectl ];
+    runtimeInputs = with pkgs; [
+      kubectl
+    ];
     text = ''
       set -euo pipefail
-      # wait until namespace file is populated
       while ! [[ -s "$1" ]]; do
         sleep 1
       done
       namespace="$(head -n1 "$1")"
-      while kubectl get ns "$namespace" 1>/dev/null 2>/dev/null; do
-        pods="$(kubectl get pods -n "$namespace" | awk '!/^NAME/{print $1}')"
-        mkdir -p "workspace/namespace-logs"
-        for pod in $pods; do
-          logfile="workspace/namespace-logs/$pod.log"
-          if ! [[ -f "$logfile" ]]; then
-            {
-              touch "$logfile" # prevents creation of to much processes
-              # wait for all containers of the pod to come online, then collect the logs
-              kubectl wait pod --all --for=condition=Ready --timeout="-1s" -n "$namespace" "$pod" 1>/dev/null 2>/dev/null
-              kubectl logs -f --all-containers=true -n "$namespace" "$pod" > "$logfile"
-            } &
-          fi
-        done
+
+      cp ./packages/log-collector.yaml ./workspace/log-collector.yaml
+      sed -i "s/@@NAMESPACE@@/''${namespace}/g" ./workspace/log-collector.yaml
+
+      kubectl apply -f ./workspace/log-collector.yaml 1>/dev/null 2>/dev/null
+
+      pod="$(kubectl get pods -o name -n "$namespace" | grep log-collector | cut -c 5-)"
+      mkdir -p ./workspace/logs
+      kubectl wait --for=condition=Ready -n "$namespace" "pod/$pod" 1>/dev/null 2>/dev/null
+      # Download and extract the logs every 3 seconds
+      while true; do
+        kubectl exec -n "$namespace" "$pod" -- /bin/bash -c "rm -f /exported-logs.tar.gz; tar zcvf /exported-logs.tar.gz /export" 1>/dev/null 2>/dev/null
+        rm -f ./workspace/logs/exported-logs.tar.gz
+        kubectl cp -n "$namespace" "$pod:/exported-logs.tar.gz" ./workspace/logs/exported-logs.tar.gz 1>/dev/null 2>/dev/null
+        tar xzvf ./workspace/logs/exported-logs.tar.gz --directory ./workspace/logs 1>/dev/null 2>/dev/null
+        sleep 3
```

Comment:
Why do we need to create tarballs continuously? I imagined that this would happen only after the test finished.

Reply:
I did not find a _good_ way to know if the test is finished or not, except maybe making an explicit download script (or giving `get-logs` CLI parameters like `start` and `download`). Also, I believe that this approach is more stable if the namespace with the log collector accidentally gets cleaned up before logs could be downloaded. If you think that this is unnecessary and has to much overhang, I'll gladly go with the "two scripts/two-part script" solution :)

Reply:
You could handle the TERM signal (or send and intercept a different signal) with logs export.

Reply:
I'm not sure if I understand what you mean correctly? I can't imagine a way to (easily) let the remote log collector send a signal to a local process and I think it should not be needed to add extra lines of code to every e2e tests go source just to tell the log collector that it is finished. I _could_ try to queue the current process list, grep the first entry that matches a `*.test` pattern and wait for its PID to stop but that seems a bit overcomplicated?

Reply:
I just split the script into two parts now, it seemed like the simplest solution to me :)